src/main/java/teammates/ui/controller/ControllerServlet.java

```diff
@@ -153,12 +153,25 @@ public final void doPost(HttpServletRequest req, HttpServletResponse resp) throw
             log.severe("Unexpected exception caught by ControllerServlet : "
                         + TeammatesException.toStringWithStackTrace(t));
             cleanUpStatusMessageInSession(req);
-            resp.sendRedirect(Const.ViewURIs.ERROR_PAGE);
+            resp.sendRedirect(appendParamsToUrlIfRequired(Const.ViewURIs.ERROR_PAGE, params));
         }
 
     }
 
     private void cleanUpStatusMessageInSession(HttpServletRequest req) {
         req.getSession().removeAttribute(Const.ParamsNames.STATUS_MESSAGES_LIST);
     }
+
+    private String appendParamsToUrlIfRequired(String baseUrl, Map<String, String[]> params) {
+        StringBuffer redirectUrl = new StringBuffer(baseUrl);
+        if (params.get(Const.ParamsNames.REGKEY) != null
+                && params.get(Const.ParamsNames.COURSE_ID) != null
+                && params.get(Const.ParamsNames.STUDENT_EMAIL) != null) {
+            redirectUrl.append('?')
+                       .append(Const.ParamsNames.REGKEY + '=' + params.get(Const.ParamsNames.REGKEY)[0] + '&')
+                       .append(Const.ParamsNames.COURSE_ID + '=' + params.get(Const.ParamsNames.COURSE_ID)[0] + '&')
+                       .append(Const.ParamsNames.STUDENT_EMAIL + '=' + params.get(Const.ParamsNames.STUDENT_EMAIL)[0]);
```

Comment:
I just took a look at `HttpRequestHelper` again and noticed a method `getValueFromParamMap`.